--- XIMEA-4.01.09/package/examples/vaViewer/main.cpp.orig	2013-11-15 15:09:58.000000000 +0100
+++ XIMEA-4.01.09/package/examples/vaViewer/main.cpp	2014-02-11 19:47:05.065480623 +0100
@@ -218,7 +218,7 @@
 				surface_width = lps->color ? lps->sbmp.w : lps->sgf.curCX;
 				surface_height = lps->color ? lps->sbmp.h : lps->sgf.curCY;
 				printf("Image size changed to %dx%d\n", surface_width, surface_height);
-				va_status = vaCreateSurfaces(va_dpy, surface_width, surface_height, VA_RT_FORMAT_YUV420, 1, &surface_id);
+				va_status = vaCreateSurfaces(va_dpy, VA_RT_FORMAT_YUV420, surface_width, surface_height, &surface_id, 1, NULL, 0);
 				CHECK_VASTATUS(va_status, "vaCreateSurfaces", lps->show=0;surface_id=VA_INVALID_SURFACE);
 				if(max_height > 0 || max_width > 0) {
 					if(max_height <= 0 || max_width > 0 && surface_width * max_height > surface_height * max_width) {
@@ -262,7 +262,7 @@
 				if (lps->dwNframes) lps->fFPS_sum = lps->fFPS_sum*tau + lps->fFPS*(1.-tau);
 				else		      lps->fFPS_sum = lps->fFPS;
 				lps->dwTime++;
-				snprintf(&win_name[0], 256, "Video FPS: %.1f; Acquisition: frm %lu skip %lu fps %.1f", lps->frame_num*1000.0/(float)lps->putsurface_time, lps->dwNframes, lps->dwLframes, (float)lps->fFPS_sum);
+				snprintf(&win_name[0], 256, "Video FPS: %.1f; Acquisition: frm %u skip %u fps %.1f", lps->frame_num*1000.0/(float)lps->putsurface_time, lps->dwNframes, lps->dwLframes, (float)lps->fFPS_sum);
 				lps->putsurface_time = 0;
 				lps->frame_num = 0;
 				XmbTextListToTextProperty(x11_display, &pwin_name, 1, XStringStyle, &winname);
